File: java-advent-of-code/src/test/java/advent/twenty_seventeen/Day12DigitalPlumberTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day12DigitalPlumberTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day12DigitalPlumber("/twenty_seventeen/Day12-DigitalPlumber-input.txt");

        assertEquals(169, dp.getPart1Answer().intValue());
        assertEquals(179, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day12DigitalPlumber("/twenty_seventeen/Day12-DigitalPlumber-example#1.txt");

        assertEquals(6, dp.getPart1Answer().intValue());
        assertEquals(2, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day12DigitalPlumberTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day12DigitalPlumberTest  extends FlatSpec with Matchers {
  "2017 Day 12 - Input" should "calculate answers" in {
    val dp = new Day12DigitalPlumber("/twenty_seventeen/Day12-DigitalPlumber-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day12DigitalPlumber("/twenty_seventeen/Day12-DigitalPlumber-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
